src/node_zlib.cc

```diff
@@ -422,6 +422,16 @@ class ZCtx : public AsyncWrap {
 
   // just pull the ints out of the args and call the other Init
   static void Init(const FunctionCallbackInfo<Value>& args) {
+    // Refs: https://github.com/nodejs/node/issues/16649
+    // Refs: https://github.com/nodejs/node/issues/14161
+    if (args.Length() == 5) {
+      fprintf(stderr,
+          "WARNING: You are likely using a version of node-tar or npm that "
+          "uses the internals of Node.js in an invalid way.\nPlease use "
```

Comment:
Micro-nit (totally not blocking): s/uses the internals of Node.js in an invalid way/is incompatible with this versino of Node.js/

Reply:
I think I prefer the current wording because it gives more of a hint on *why* the error is occurring. That the versions are incompatible kind of follows from the rest of the message, right?

Reply:
I guess I'm trying to avoid "uses the internals of Node.js in an invalid way" as it seems likely to be seen as throwing a bit of shade at npm. But I may be overthinking it.

Beyond that, my thinking was that the user needs to know the versions are incompatible, and why (or whose responsible for the situation) are secondary. 

All that said, I'm fine with the current wording.

Reply:
@trott yup, updated!